squad/core/models.py

```diff
@@ -1058,20 +1058,31 @@ def get_fixes(self):
     def get_exceeded_thresholds(self):
         # Return a list of all (threshold, metric) objects for those
         # thresholds that were exceeded by corresponding metrics.
+        environments_qs = self.build.project.environments.all()
         thresholds_exceeded = []
         if self.has_metrics:
             test_runs = self.build.test_runs.all()
             suites = Suite.objects.filter(test__test_run__build=self.build)
+            thresholds = MetricThreshold.objects.filter(
+                environment__in=environments_qs.values_list('id', flat=True)
+            )
+            thresholds_names = thresholds.values_list('name', flat=True)
             for metric in Metric.objects.filter(
                     Q(test_run__in=test_runs) | Q(suite__in=suites),
-                    name__in=self.build.project.metricthreshold_set.values_list('name', flat=True)):
-                for threshold in self.build.project.metricthreshold_set.all():
+                    name__in=thresholds_names):
+                for threshold in thresholds:
```

Comment:
OK, This is a bit confusing when adding environments, but I think we can solve that changing very little code:

Although you added environment to MetricThreshold, you're still filtering metrics that exceeded in a project's threshold.

What you need to do is check if the Metric's TestRun's environment matches the environment in the threshold. something like that:
```
for metric in Metric.objects.filter(...): # this filtering is OK
    for threshold in thresholds:
        if metric.test_run.environment_id != threshold.environment_id:
            continue # this threshold doesn't apply to this metric 
```

But this will imply that you should have `thresholds` an iterator of objects, instead of just a list of metric threshold names.

Does it make sense?